modules/shared.py

```diff
@@ -159,6 +159,7 @@ def str2bool(v):
 # API
 parser.add_argument('--api', action='store_true', help='Enable the API extension.')
 parser.add_argument('--public-api', action='store_true', help='Create a public URL for the API using Cloudfare.')
+parser.add_argument('--thread-safe', action='store_true', help='Queue requests to API. Useful if multiple clients accessing API.')
```

Comment:
I wonder if this should be handled in the [params ](https://github.com/oobabooga/text-generation-webui/blob/main/docs/Extensions.md#params-dictionary) for the api extension instead of an argument like this? Might be confusing since it only affects the non-Gradio API.

Reply:
It's fine like this, the gradio API has been removed and this is the only API